lib/app/utils/measure_operations.rb

```diff
@@ -11,9 +11,34 @@ def evaluate_measure(measure_id, params = {})
       @client.get "Measure/#{measure_id}/$evaluate-measure#{params_string}", @client.fhir_headers(format: FHIR::Formats::ResourceFormat::RESOURCE_JSON)
     end
 
-    def submit_data
-      # TODO
-      nil
+    def submit_data(measure_id, patient_id, period_start, period_end, patient_resources)
+      measure_report = FHIR::STU3::MeasureReport.new
+      measure_report.type = 'individual'
+      measure_report.patient = FHIR::STU3::Reference.new
+      measure_report.patient.reference = "Patient/#{patient_id}"
+      measure_report.measure = FHIR::STU3::Reference.new
+      measure_report.measure.reference = "Measure/#{measure_id}"
+      measure_report.period = FHIR::STU3::Period.new
+      measure_report.period.start = period_start
+      measure_report.period.end = period_end
+
+      parameters = FHIR::STU3::Parameters.new
+      measure_report_param = FHIR::STU3::Parameters::Parameter.new
+      measure_report_param.name = 'measure-report'
+      measure_report_param.resource = measure_report
+      parameters.parameter.push(measure_report_param)
+      patient_resources.each do |r|
+        resource_param = FHIR::STU3::Parameters::Parameter.new
+        resource_param.name = 'resource'
+        resource_param.resource = r
+        parameters.parameter.push(resource_param)
```

Comment:
NIT: I'm not 100% sure about this, or what the actual best Ruby way to do this is, but is there a way we can make use of `from_json` to create the resource all at once rather than having many individual statements?

Reply:
I'm not entirely clear on what you mean. The only JSON we have is a `Bundle` that's already deserialized into Ruby at this point. All of these things here exist to create the [FHIR Parameters](https://www.hl7.org/fhir/parameters.html) object, which includes creating a new `MeasureReport` (lines 15-23), creating a `parameter` element for that `MeasureReport` (lines 25-29), and creating a `parameter` element for each resource (lines 30-34).

Reply:
What I mean is can we create these resources without doing all these individual statements, like this:

``` Ruby
measure_report = FHIR::STU3::Json.from_json({
  resourceType: 'MeasureReport',
  type: 'individual',
  patient: {
    reference: "Patient/#{patient_id}"
  },
  measure: {
    reference: "Measure/#{measure_id}"
  },
  period: {
    start: period_start,
    end: period_end
  }
})
```

Or I guess my question is, is it cleaner/better to do it the above way rather than instantiating everything individually?

Reply:
Might be worth it to have @okeefm weigh in on this with his immeasurable Ruby knowledge

Reply:
The only problem I see with something like that is that technically speaking, that hash isn't JSON. So you'd have to do something like `measure_report = FHIR::STU3::Json.from_json({ resourceType: 'MeasureReport', ...}.to_json)`, which seems overly elaborate to me (though other Rubyists can chime in too).

That being said, there may be a way to create a FHIR object from a `Hash`, which I'm going to look into a bit.

Reply:
Yeah, there's a `from_hash` method on FHIR objects in FHIR Models, though the only catch is you have to call it on an instance, not the class. So you could do something like 
```
measure_report = FHIR::STU3::MeasureReport.new.from_hash({
  resourceType: 'MeasureReport',
  type: 'individual',
  patient: {
    reference: "Patient/#{patient_id}"
  },
  measure: {
    reference: "Measure/#{measure_id}"
  },
  period: {
    start: period_start,
    end: period_end
  }
})
```

Reply:
do we think the above has any benefit over the way it is now, from a code styling perspective or otherwise?

Reply:
It definitely reads as more Javascript-y this way. Whether it's more or less idiomatic Ruby I'm not sure. Personally, I do find some readability improvements from the hash method, but I don't think either way is "wrong" per se.

Reply:
@okeefm do you think we should leave it as-is, switch to the `from_hash` method, or is there another way you would do it?

Reply:
@tstrass I think it's up to you, I don't have strong enough feelings beyond what I listed above (I find the `from_hash` method slightly more readable at a glance, but that may just be because I do ~any~ too much JS dev).

Reply:
I know it's not wrong as-is, so this is basically just a style question, but style matters.